File: app/src/main/java/de/saschahlusiak/frupic/db/FrupicDBOpenHandler.java
package de.saschahlusiak.frupic.db;
import android.content.Context;
import android.database.sqlite.SQLiteDatabase;
import android.database.sqlite.SQLiteOpenHelper;
import android.util.Log;

public class FrupicDBOpenHandler extends SQLiteOpenHelper {
	private static final String DATABASE_NAME = "frupic.db";
	private static final int DATABASE_VERSION = 2;
	private static final String tag = FrupicDBOpenHandler.class.getSimpleName();

	public FrupicDBOpenHandler(Context context) {
		super(context, DATABASE_NAME, null, DATABASE_VERSION);
	}

	@Override
	public void onCreate(SQLiteDatabase db) {
		db.execSQL(FrupicDB.CREATE_TABLE);

	}

	@Override
	public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
		try {
			if (oldVersion < 2) {
				db.execSQL(FrupicDB.UPGRADE_FROM_1);
			}
		}catch (Exception e) {
			e.printStackTrace();
			db.execSQL(FrupicDB.DROP_TABLE);
			onCreate(db);
		}
	}

}

File: app/src/main/java/de/saschahlusiak/frupic/db/FrupicDBOpenHandler.kt
package de.saschahlusiak.frupic.db

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class FrupicDBOpenHandler(context: Context?) : SQLiteOpenHelper(context, DATABASE_NAME, null, DATABASE_VERSION) {
    override fun onCreate(db: SQLiteDatabase) {
        db.execSQL(FrupicDB.CREATE_TABLE)
    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        try {
            if (oldVersion < 2) {
                db.execSQL(FrupicDB.UPGRADE_FROM_1)
            }
        } catch (e: Exception) {
            e.printStackTrace()
            db.execSQL(FrupicDB.DROP_TABLE)
            onCreate(db)
        }
    }

    companion object {
        private const val DATABASE_NAME = "frupic.db"
        private const val DATABASE_VERSION = 2
        private val tag = FrupicDBOpenHandler::class.java.simpleName
    }
}